### ISSUE TITLE
geoserver - wms 1.3.0 fails to serve rasters in their native projection

### ISSUE THREAD

**Issue by fphg** [georchestra/georchestra#911]
Hello
Big issue concerning GS with EPSG:2154 datas on GeoServer 2.5.4

This getmap works : 2154 raster data reproj to 3857 using WMS 1.3.0
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.3.0&SLD_VERSION=1.1.0&REQUEST=GetMap&STYLES=&CRS=EPSG%3A3857&BBOX=-519469,6112666,-519310,6112825&WIDTH=532&HEIGHT=532

This getmap works : 2154 raster data to 2154 using WMS 1.1.0
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.1.0&SLD_VERSION=1.1.0&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=111423,6780205,131755,6800538&WIDTH=532&HEIGHT=532

This getmap fails : 2154 raster data to 2154 using WMS 1.3.0
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.3.0&SLD_VERSION=1.1.0&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=111423,6780205,131755,6800538&WIDTH=532&HEIGHT=532

Error : 
      Error rendering coverage on the fast path
org.opengis.geometry.MismatchedReferenceSystemException: The coordinate reference system must be the same for all objects.
The coordinate reference system must be the same for all objects.

**Comment by fphg** [comment]
traces here
http://sourceforge.net/p/geoserver/mailman/message/32791363/

and here
http://jira.codehaus.org/browse/GEOS-6651

**Comment by fvanderbiest** [comment]
Funnily http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.3&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=111423,6780205,131755,6800538&WIDTH=532&HEIGHT=532 works ... (I just changed 1.3.0 into 1.3)

**Comment by fvanderbiest** [comment]
The test image bounding box is:
```
left = 129000
bottom = 6798000
right = 130000
top = 6799000
```

Working request on full bbox:
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.1.0&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=129000,6798000,130000,6799000&WIDTH=532&HEIGHT=532

Request on full bbox (with coordinated swapped, as requested by the WMS 1.3.0 spec for EPSG:2154), but this time, with a blank image as result (no more XML error):
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=1.3.0&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=6798000,129000,6799000,130000&WIDTH=532&HEIGHT=532

In light of the WMS 1.3.0 spec, which says: "If a request contains **a BBOX whose area does not overlap** at all with the <BoundingBox> element in the service metadata for the requested layer, the server shall return empty content (that is, **a blank map** or an graphic element file with no elements) for that map. Any features that are partly or entirely contained in the Bounding Box shall be returned in the appropriate format." ... we can guess that the server misinterprets the given coordinates.

**Comment by hsquividant** [comment]
I'm not sure that in your case 1.3 = 1.3.0. Most probably GS replace it by default version (1.1.0?)
Trying with any wacky WMS version works too !
http://sdi.georchestra.org/geoserver/test2/ows?SERVICE=WMS&LAYERS=ortho&EXCEPTIONS=XML&FORMAT=image%2Fpng&TRANSPARENT=TRUE&VERSION=29.29&REQUEST=GetMap&STYLES=&CRS=EPSG%3A2154&BBOX=111423,6780205,131755,6800538&WIDTH=532&HEIGHT=532

**Comment by fvanderbiest** [comment]
> Most probably GS replace it by default version (1.1.0?)

That's what I was thinking too ...

**Comment by fvanderbiest** [comment]
I've checked the ```coverage.xml``` files generated by a 2.5.4 instance against those generated by a 2.3.3 instance (which did not have the issue). There are the same ...

### EPSG:2154 Raster in GeoServer 2.3.2

Corner Coordinates:
```
Upper Left  (  580486.185, 7044189.085) (  1d19' 6.55"E, 50d29' 1.08"N)
Lower Left  (  580486.185, 6854789.085) (  1d22'24.15"E, 48d46'55.61"N)
Upper Right (  796686.185, 7044189.085) (  4d21'37.47"E, 50d29'15.29"N)
Lower Right (  796686.185, 6854789.085) (  4d18'57.59"E, 48d47' 9.38"N)
Center      (  688586.185, 6949489.085) (  2d50'31.39"E, 49d38'39.97"N)
```

coverage.xml 
```xml
  <nativeCRS class="projected">PROJCS[&quot;unnamed&quot;,
  GEOGCS[&quot;GCS Name = GCS_RGF_1993&quot;,
    DATUM[&quot;Reseau Geodesique Francais 1993&quot;,
      SPHEROID[&quot;GRS 1980&quot;, 6378137.0, 298.257222101, AUTHORITY[&quot;EPSG&quot;,&quot;7019&quot;]],
      TOWGS84[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
      AUTHORITY[&quot;EPSG&quot;,&quot;6171&quot;]],
    PRIMEM[&quot;Greenwich&quot;, 0.0, AUTHORITY[&quot;EPSG&quot;,&quot;8901&quot;]],
    UNIT[&quot;degree&quot;, 0.017453292519943295],
    AXIS[&quot;Geodetic longitude&quot;, EAST],
    AXIS[&quot;Geodetic latitude&quot;, NORTH]],
  PROJECTION[&quot;Lambert_Conformal_Conic_2SP&quot;],
  PARAMETER[&quot;central_meridian&quot;, 3.0],
  PARAMETER[&quot;latitude_of_origin&quot;, 46.5],
  PARAMETER[&quot;standard_parallel_1&quot;, 49.0],
  PARAMETER[&quot;false_easting&quot;, 700000.0],
  PARAMETER[&quot;false_northing&quot;, 6600000.0],
  PARAMETER[&quot;scale_factor&quot;, 1.0],
  PARAMETER[&quot;standard_parallel_2&quot;, 44.0],
  UNIT[&quot;m&quot;, 1.0],
  AXIS[&quot;Easting&quot;, EAST],
  AXIS[&quot;Northing&quot;, NORTH]]</nativeCRS>
  <srs>EPSG:2154</srs>
  <nativeBoundingBox>
    <minx>580486.1849871671</minx>
    <maxx>796686.1849871671</maxx>
    <miny>6854789.084880707</miny>
    <maxy>7044189.084880707</maxy>
    <crs class="projected">EPSG:2154</crs>
  </nativeBoundingBox>
  <latLonBoundingBox>
    <minx>1.3184872668830656</minx>
    <maxx>4.360407938162931</maxx>
    <miny>48.78211361447407</miny>
    <maxy>50.49495208981582</maxy>
    <crs>GEOGCS[&quot;WGS84(DD)&quot;,
  DATUM[&quot;WGS84&quot;,
    SPHEROID[&quot;WGS84&quot;, 6378137.0, 298.257223563]],
  PRIMEM[&quot;Greenwich&quot;, 0.0],
  UNIT[&quot;degree&quot;, 0.017453292519943295],
  AXIS[&quot;Geodetic longitude&quot;, EAST],
  AXIS[&quot;Geodetic latitude&quot;, NORTH]]</crs>
  </latLonBoundingBox>
  <projectionPolicy>FORCE_DECLARED</projectionPolicy>
```


### EPSG:2154 Raster in GeoServer 2.5.4
```
Corner Coordinates:
Upper Left  (  129000.000, 6799000.000) 
Lower Left  (  129000.000, 6798000.000) 
Upper Right (  130000.000, 6799000.000) 
Lower Right (  130000.000, 6798000.000) 
Center      (  129500.000, 6798500.000)
```
coverage.xml 
```xml
  <nativeCRS class="projected">PROJCS[&quot;RGF93 / Lambert-93&quot;,
  GEOGCS[&quot;RGF93&quot;,
    DATUM[&quot;Reseau Geodesique Francais 1993&quot;,
      SPHEROID[&quot;GRS 1980&quot;, 6378137.0, 298.257222101, AUTHORITY[&quot;EPSG&quot;,&quot;7019&quot;]],
      TOWGS84[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
      AUTHORITY[&quot;EPSG&quot;,&quot;6171&quot;]],
    PRIMEM[&quot;Greenwich&quot;, 0.0, AUTHORITY[&quot;EPSG&quot;,&quot;8901&quot;]],
    UNIT[&quot;degree&quot;, 0.017453292519943295],
    AXIS[&quot;Geodetic longitude&quot;, EAST],
    AXIS[&quot;Geodetic latitude&quot;, NORTH],
    AUTHORITY[&quot;EPSG&quot;,&quot;4171&quot;]],
  PROJECTION[&quot;Lambert_Conformal_Conic_2SP&quot;, AUTHORITY[&quot;EPSG&quot;,&quot;9802&quot;]],
  PARAMETER[&quot;central_meridian&quot;, 3.0],
  PARAMETER[&quot;latitude_of_origin&quot;, 46.5],
  PARAMETER[&quot;standard_parallel_1&quot;, 49.0],
  PARAMETER[&quot;false_easting&quot;, 700000.0],
  PARAMETER[&quot;false_northing&quot;, 6600000.0],
  PARAMETER[&quot;scale_factor&quot;, 1.0],
  PARAMETER[&quot;standard_parallel_2&quot;, 44.0],
  UNIT[&quot;m&quot;, 1.0],
  AXIS[&quot;Easting&quot;, EAST],
  AXIS[&quot;Northing&quot;, NORTH],
  AUTHORITY[&quot;EPSG&quot;,&quot;2154&quot;]]</nativeCRS>
  <srs>EPSG:2154</srs>
  <nativeBoundingBox>
    <minx>129000.0</minx>
    <maxx>130000.0</maxx>
    <miny>6798000.0</miny>
    <maxy>6799000.0</maxy>
    <crs class="projected">EPSG:2154</crs>
  </nativeBoundingBox>
  <latLonBoundingBox>
    <minx>-4.674320311383806</minx>
    <maxx>-4.659664751011645</maxx>
    <miny>48.03250059405376</miny>
    <maxy>48.042329490580045</maxy>
    <crs>GEOGCS[&quot;WGS84(DD)&quot;,
  DATUM[&quot;WGS84&quot;,
    SPHEROID[&quot;WGS84&quot;, 6378137.0, 298.257223563]],
  PRIMEM[&quot;Greenwich&quot;, 0.0],
  UNIT[&quot;degree&quot;, 0.017453292519943295],
  AXIS[&quot;Geodetic longitude&quot;, EAST],
  AXIS[&quot;Geodetic latitude&quot;, NORTH]]</crs>
  </latLonBoundingBox>
  <projectionPolicy>FORCE_DECLARED</projectionPolicy>
```

**Comment by fvanderbiest** [comment]
I just commented the relevant issue in GeoServer issue tracker http://jira.codehaus.org/browse/GEOS-6651?focusedCommentId=363112&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-363112

**Comment by fvanderbiest** [comment]
Jukka says he does not witness the issue with the same raster on GeoServer 2.7-beta ...

**Comment by fvanderbiest** [comment]
I tested on a non-geOrchestra geoserver 2.5.4 with it's default data dir, same issue.

**Comment by fvanderbiest** [comment]
The issue is gone on a vanilla 2.6.2 geoserver.

**Comment by fvanderbiest** [comment]
And 2.6 beta is affected too.

**Comment by fvanderbiest** [comment]
Could not find anything related in the bug tracker https://jira.codehaus.org/browse/GEOS-6598?jql=project%20%3D%20GEOS%20AND%20resolution%20%3D%20Fixed%20AND%20fixVersion%20%3D%20%222.6-RC1%22%20ORDER%20BY%20issuetype%20ASC%2C%20priority%20DESC

The diff between 2.6 beta and rc1 is available here: https://gist.github.com/fvanderbiest/7cd9bcb7ef5e1c9fbaff (if someone wants to have a look)

**Comment by fvanderbiest** [comment]
For those who are stuck with this issue, I have built with docker a simple POC showing that a regular geoserver 2.6.2 can run behind the security proxy and intercept the security headers (thanks to an older documentation made by Fabrice and Gaston).

The diff is here: https://github.com/georchestra/docker/commit/6ae2cf856c0933ccc9d510b76048a0ad5a7467e4

Though I must admit I did not test it exhaustively...

**Comment by fphg** [comment]
Hi
2.6.2 services work well, but one can't use the admin UI because sessions aren't working and we don't know why. We tried a vanilla 2.6.2 geoserver and get the same results. Session mechanism can be tested against multi-pages displays (for example : demo > src list).
However, sessions are working again if bypassing the 14.12 security proxy.

**Comment by fvanderbiest** [comment]
This 2.5.4 "georchestra" geoserver does not exhibit the behavior described here: EPSG:2154 data displayed in an EPSG:2154 map with WMS 1.3.0 ...

http://www.cigalsace.org/gs-public/SHD1/wms?service=WMS&version=1.3.0&request=GetMap&layers=SHD1:SHD1_GR_6_M_LVIB1C_24_0001_H&styles=&bbox=978298.1105282841,6742178.934515951,995434.0545338725,6765298.188821313&width=379&height=512&srs=EPSG:2154&format=application/openlayers

**Comment by fvanderbiest** [comment]
Closing the issue, since we cannot fix GeoServer 2.5.4.
Recommended upgrade: GeoServer standalone or 2.8 in geOrchestra 15.12
